pkg/bearertoken/grpc_test.go

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package bearertoken
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+func TestBearerTokenInterceptors(t *testing.T) {
+	tests := []struct {
+		name        string
+		ctx         context.Context
+		expectedErr string
+		expectedMD  metadata.MD
+	}{
+		{
+			name:        "no token in context",
+			ctx:         context.Background(),
+			expectedErr: "",
+			expectedMD:  nil, // Expecting no metadata
+		},
+		{
+			name:        "token in context",
+			ctx:         ContextWithBearerToken(context.Background(), "test-token"),
+			expectedErr: "",
+			expectedMD:  metadata.MD{Key: []string{"test-token"}},
+		},
+		{
+			name:        "multiple tokens in metadata",
+			ctx:         metadata.NewIncomingContext(context.Background(), metadata.MD{Key: []string{"token1", "token2"}}),
+			expectedErr: "malformed token: multiple tokens found",
+		},
+		{
+			name:        "valid token in metadata",
+			ctx:         metadata.NewIncomingContext(context.Background(), metadata.MD{Key: []string{"valid-token"}}),
+			expectedErr: "",
+			expectedMD:  metadata.MD{Key: []string{"valid-token"}}, // Valid token setup
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// Unary interceptor test
+			unaryInterceptor := NewUnaryClientInterceptor()
+			unaryInvoker := func(ctx context.Context, _ string, _, _ any, _ *grpc.ClientConn, _ ...grpc.CallOption) error {
+				md, ok := metadata.FromOutgoingContext(ctx)
+				if test.expectedMD == nil {
+					require.False(t, ok) // There should be no metadata in this case
+				} else {
+					require.True(t, ok)
+					assert.Equal(t, test.expectedMD, md)
+				}
+				return nil
+			}
+			err := unaryInterceptor(test.ctx, "method", nil, nil, nil, unaryInvoker)
+			if test.expectedErr == "" {
+				require.NoError(t, err)
+			} else {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), test.expectedErr)
+			}
+
+			// Stream interceptor test
+			streamInterceptor := NewStreamClientInterceptor()
```

Comment:
you seem to only have tests for client interceptors, what about server interceptors?